process/coordinator/process.go

```diff
@@ -671,12 +671,12 @@ func (tc *transactionCoordinator) CreateMarshalizedData(body *block.Body) map[st
 
 	for i := 0; i < len(body.MiniBlocks); i++ {
 		miniblock := body.MiniBlocks[i]
-		receiverShardId := miniblock.ReceiverShardID
-		if receiverShardId == tc.shardCoordinator.SelfId() {
+		if miniblock.SenderShardID != tc.shardCoordinator.SelfId() ||
```

Comment:
this is wrong -> effectively you do not marshall any miniblock.

Reply:
It is ok, we want to broadcast only miniblocks created in self shard with destination in other shards